src/CMakeLists.txt

```diff
@@ -1,107 +1,78 @@
 # Must be included before CMAKE_INSTALL_INCLUDEDIR is used.
 include(GNUInstallDirs)
-set(${PROJECT_NAME}_installables "")
 
+add_library(secp_precomputed_objs OBJECT EXCLUDE_FROM_ALL
+  precomputed_ecmult.c
+  precomputed_ecmult_gen.c
+)
+
+# Add objects explicitly rather than linking to the object libs to keep them
+# from being exported.
+add_library(secp256k1 secp256k1.c $<TARGET_OBJECTS:secp_precomputed_objs>)
+
+add_library(secp_asm INTERFACE)
 if(SECP256K1_ASM STREQUAL "arm")
-  add_library(common OBJECT
+  add_library(secp_asm_arm OBJECT EXCLUDE_FROM_ALL)
+  target_sources(secp_asm_arm PUBLIC
```

Comment:
Why `PUBLIC`? If it was unintentional then adding of the `target_sources` seems unneeded, no?

Reply:
Otherwise it ends up missing as part of the interface and link fails. Is there some trick I'm missing?